jwk.go

```diff
@@ -348,6 +362,35 @@ func (s *JSONWebKeySet) Key(kid string) []JSONWebKey {
 	return keys
 }
 
+func (s *JSONWebKeySet) UnmarshalJSON(data []byte) (err error) {
```

Comment:
👋 @mcpherrinm I'm happy to find you here :)
We are trying to update go-jose from 4.0.2 to 4.0.3 in the [SPIRE](https://github.com/spiffe/spire/pull/5288) and [go-spiffe](https://github.com/spiffe/go-spiffe/pull/292) projects, but unit tests are failing with 4.0.3.
Looking at this PR, I can see the introduction of this function, which influences the way that a `json.Unmarshal` works when unmarshaling a struct with the `jose.JSONWebKeySet` type embedded. This is because now it can be called during the decoding process, as the decoder [now encounters this Unmarshaler that can be called](https://github.com/golang/go/blob/master/src/encoding/json/decode.go#L479) but the result is that [the struct that holds the `jose.JSONWebKeySet` struct](https://github.com/spiffe/spire/blob/main/pkg/common/bundleutil/types.go#L12-L16) does not get the fields properly populated (e.g. the RefreshHint field).
I'm not quite sure if we are doing something wrong in our side. To see if this change is related with the failure, I deleted this function from the 4.0.3 code and the parsing / decoding works as expected, with the tests passing.
I would really appreciate some help to find a solution here. Thanks!

Reply:
please file an issue

Reply:
Filed #136.